packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -128,16 +145,92 @@ class TextMapPropagator {
     }
   }
 
+  _injectB3SingleHeader (spanContext, carrier) {
+    const hasB3SingleHeader = this._hasPropagationStyle('inject', 'b3 single header')
+    if (!hasB3SingleHeader) return null
+
+    const traceId = spanContext._traceId.toString(16)
+    const spanId = spanContext._spanId.toString(16)
+    const sampled = spanContext._sampling.priority >= AUTO_KEEP ? '1' : '0'
+
+    carrier[b3HeaderKey] = `${traceId}-${spanId}-${sampled}`
+    if (spanContext._parentId) {
+      carrier[b3HeaderKey] += '-' + spanContext._parentId.toString(16)
+    }
+  }
+
   _injectTraceparent (spanContext, carrier) {
-    if (!this._config.experimental.traceparent) return
+    if (!this._hasPropagationStyle('inject', 'tracecontext')) return
+
+    const {
+      _sampling: { priority, mechanism },
+      _tracestate: ts,
+      _trace: { origin, tags }
+    } = spanContext
+
     carrier[traceparentKey] = spanContext.toTraceparent()
+
+    ts.forVendor('dd', state => {
+      state.set('s', priority)
+      if (mechanism) {
+        state.set('t.dm', mechanism)
+      }
+
+      if (typeof origin === 'string') {
+        const originValue = origin
+          .replace(tracestateOriginFilter, '_')
+          .replace(/[\x3d]/g, '~')
+
+        state.set('o', originValue)
+      }
+
+      for (const key in tags) {
+        if (!tags[key] || !key.startsWith('_dd.p.')) continue
+
+        const tagKey = 't.' + key.slice(6)
+          .replace(tracestateTagKeyFilter, '_')
+
+        const tagValue = tags[key]
+          .toString()
+          .replace(tracestateTagValueFilter, '_')
+          .replace(/[\x3d]/g, '~')
+
+        state.set(tagKey, tagValue)
+      }
+    })
+
+    carrier.tracestate = ts.toString()
+  }
+
+  _hasPropagationStyle (mode, name) {
+    return this._config.tracePropagationStyle[mode].includes(name)
   }
 
   _extractSpanContext (carrier) {
-    return this._extractDatadogContext(carrier) ||
-      this._extractTraceparentContext(carrier) ||
-      this._extractB3Context(carrier) ||
-      this._extractSqsdContext(carrier)
+    for (const extractor of this._config.tracePropagationStyle.extract) {
+      let spanContext = null
+      switch (extractor) {
```

Comment:
This seems to only pick the first one, how does it fallback if `null` is returned by an extract method?

Reply:
If the `spanContext !== null` check below fails it will continue to the next iteration. The switch just routes to the different extractors and then sets the `spanContext` value. _After_ the switch it checks if it _got_ a value and decides if it should return or continue. Seemed cleaner to me than duplicating returns across every branch.